Arrow (dependency of Spark) fails to build on Arm64

Just logging this here for future reference, it failed during the `postprocessing` step:

```
  -- System processor: aarch64
  -- Performing Test CXX_SUPPORTS_ARMV8_ARCH
  -- Performing Test CXX_SUPPORTS_ARMV8_ARCH - Failed
  -- Arrow build warning level: PRODUCTION
  CMake Error at cmake_modules/SetupCxxFlags.cmake:339 (message):
    Unsupported arch flag: -march=.
  Call Stack (most recent call first):
    CMakeLists.txt:100 (include)
  -- Configuring incomplete, errors occurred!
  See also "/tmp/eb-dov8gf9q/pip-req-build-x1zhcivt/build/temp.linux-aarch64-3.8/CMakeFiles/CMakeOutput.log".
  See also "/tmp/eb-dov8gf9q/pip-req-build-x1zhcivt/build/temp.linux-aarch64-3.8/CMakeFiles/CMakeError.log".
  error: command 'cmake' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyarrow
Failed to build pyarrow
ERROR: Could not build wheels for pyarrow which use PEP 517 and cannot be installed directly
 (at easybuild/tools/run.py:537 in parse_cmd_output)
```

This should be solved in newer versions, see:
https://github.com/apache/arrow/pull/7982

@ocaisa backported that fix for our version in https://github.com/easybuilders/easybuild-easyconfigs/pull/12640

This has to be included in the build script.

--- a/EESSI-pilot-install-software.sh
+++ b/EESSI-pilot-install-software.sh
@@ -322,6 +322,15 @@ fail_msg="Installation of OSU-Micro-Benchmarks failed, that's unexpected..."
 $EB OSU-Micro-Benchmarks-5.6.3-gompi-2020a.eb -r
 check_exit_code $? "${ok_msg}" "${fail_msg}"
 
+# Arrow 0.17.1 (dependency of Spark) needs a fix on aarch64
+# may not be necessary for newer versions:
+# https://github.com/apache/arrow/pull/7982
+echo ">> Installing Arrow 0.17.1..."
+ok_msg="Arrow installed, enjoy!"
+fail_msg="Installation of Arrow failed..."
+$EB --from-pr 12640 Arrow-0.17.1-foss-2020a-Python-3.8.2.eb -r
+check_exit_code $? "${ok_msg}" "${fail_msg}"
+
 echo ">> Installing Spark 3.1.1..."
 ok_msg="Spark installed, set off the fireworks!"
 fail_msg="Installation of Spark failed, no fireworks this time..."

